src/main/knn/KnnGraphTester.java

```diff
@@ -330,6 +338,13 @@ private void run(String... args) throws Exception {
             throw new IllegalArgumentException("-numMergeThread should be >= 1");
           }
           break;
+        case "-parentJoin":
+          if (iarg == args.length - 1) {
+            throw new IllegalArgumentException("-parentJoin requires a following Path for parentJoinMetaFile");
+          }
+          parentJoinMetaFile = Paths.get(args[++iarg]);
```

Comment:
Instead of separate `boolean parentJoin` you could just use `parentJoinMetaFile != null` to know when to do parent/join ...

Reply:
It am currently only using a single input param, `-parentJoin <parentJoinMetaFile>` for KnnGraphTester, which we would need either way. Storing this condition in a boolean is a convenience of not having to add the null check at all the places where this is needed.